test/cqlpy/test_vector_similarity.py

```diff
@@ -0,0 +1,311 @@
+# Copyright 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+
+import pytest
+from .util import new_test_table, is_scylla
+from cassandra.protocol import InvalidRequest
+from math import sqrt, isclose
+
+
+###############################################################################
+# Tests for vector search related functions
+###############################################################################
+
+similarity_functions = {"cosine", "euclidean", "dot_product"}
+
+
+@pytest.fixture
```

Comment:
please use `@pytest.fixture(scope="module")` to make it clearer that this is a file-local fixture.